cmake/CPM.cmake

```diff
@@ -750,19 +752,28 @@ endfunction()
 
 # downloads a previously declared package via FetchContent and exports the variables
 # `${PACKAGE}_SOURCE_DIR` and `${PACKAGE}_BINARY_DIR` to the parent scope
-function(cpm_fetch_package PACKAGE)
+function(cpm_fetch_package PACKAGE populated)
+  set(${populated}
+      FALSE
+      PARENT_SCOPE
+  )
```

Comment:
It's a personal preference, but could you move this set into the `else()` case for the `NOT ${lower_case_name}_POPULATED` case below? IMO it's a bit more readable as the "return" equivalent is called only once at the end of the function.

Reply:
I would have but there is:
https://github.com/cpm-cmake/CPM.cmake/blob/c5cb85b2f1eae7e23fdb33bb2892d034a0e46404/cmake/CPM.cmake#L754-L757

Wouldn't it cause problem in case of dry run?

Reply:
Good point, I missed that. In general I wouldn't worry about the dry run as it's currently only used for one old unit test and may be removed completely soon. Let's keep it as is then for now and update the statement in the future.